File: c/armstrong-numbers/armstrong_numbers.c
#include "armstrong_numbers.h"
#include <math.h>
#include <stdio.h>


bool is_armstrong_number(int candidate)
{
	/* Count the number of digits */
	int num_digits =  candidate > 0 ? log10(candidate) + 1 : 0;

	/* Check if Armstrong number by performing calculation */
	int sum_power = 0;
	int _candidate = candidate;
	while (_candidate > 0)
	{
		sum_power += pow(_candidate % 10, num_digits);
		_candidate /= 10;
	}

	return (sum_power == candidate);
}

File: python/armstrong-numbers/armstrong_numbers.py
from math import log10


def is_armstrong_number(number):
    num_digits = 0 if number <= 0 else 1 + int(log10(number))
    armstrong_sum = 0
    _number = number
    while _number > 0:
        _number, d = _number // 10, _number % 10
        armstrong_sum += d**num_digits

    return number == armstrong_sum
